proxysql_node_monitor

```diff
@@ -62,15 +62,32 @@ update_cluster(){
       ws_ip=$(echo $i | cut -d':' -f1)
       ws_port=$(echo $i | cut -d':' -f2)
       echo "Node ${i} does not exists in cluster membership!" >> $ERR_FILE
-      proxysql_exec "DELETE FROM mysql_servers WHERE hostname='$ws_ip' and port=$ws_port;"
-      check_cmd $? "Cannot remove Percona XtraDB Cluster node $ws_ip:$ws_port from ProxySQL database, Please check proxysql credentials"
+      proxysql_exec "UPDATE mysql_servers set status = 'OFFLINE_HARD' WHERE hostgroup_id = $HOSTGROUP_ID and hostname='$ws_ip' and port=$ws_port;"
+      check_cmd $? "Cannot update Percona XtraDB Cluster node $ws_ip:$ws_port (hostgroup $HOSTGROUP_ID) to ProxySQL database, Please check proxysql credentials"
       proxysql_exec "LOAD MYSQL SERVERS TO RUNTIME; SAVE MYSQL SERVERS TO DISK;"
-      echo "Removed ${i} node from ProxySQL database." >> $ERR_FILE 
+      echo "${i} node set to OFFLINE_HARD status to ProxySQL database." >> $ERR_FILE 
     else
       CHECK_STATUS=1 
     fi
   done
-  proxysql_exec "UPDATE SCHEDULER SET arg3=${#wsrep_address[@]} WHERE id=10;LOAD SCHEDULER TO RUNTIME;SAVE SCHEDULER TO DISK;"
+
+  for i in "${wsrep_address[@]}"; do
+    if [[ ! " ${current_hosts[@]} " == " ${i} " ]]; then
+      ws_ip=$(echo $i | cut -d':' -f1)
+      ws_port=$(echo $i | cut -d':' -f2)
+      ws_status=$(echo `proxysql_exec "SELECT status from mysql_servers WHERE hostgroup_id = $HOSTGROUP_ID and hostname='$ws_ip' and port=$ws_port"`)
+      echo "Cluster node (${i}) current status '$ws_status' in ProxySQL database!" >> $ERR_FILE 
+      if [ "$ws_status" == "OFFLINE_HARD" ]; then
+	      proxysql_exec "UPDATE mysql_servers set status = 'OFFLINE_SOFT' WHERE hostgroup_id = $HOSTGROUP_ID and hostname='$ws_ip' and port=$ws_port;"
+        check_cmd $? "Cannot update Percona XtraDB Cluster node $ws_ip:$ws_port (hostgroup $HOSTGROUP_ID) to ProxySQL database, Please check proxysql credentials" 
+        proxysql_exec "LOAD MYSQL SERVERS TO RUNTIME; SAVE MYSQL SERVERS TO DISK;"
+        echo "${i} node set to OFFLINE_SOFT status to ProxySQL database." >> $ERR_FILE
+      fi
+    else
+      CHECK_STATUS=1
+    fi
+  done
+  proxysql_exec "UPDATE SCHEDULER SET arg4=${#wsrep_address[@]} WHERE id=$SCHEDULER_ID;LOAD SCHEDULER TO RUNTIME;SAVE SCHEDULER TO DISK;"
```

Comment:
I changed the arg index because in the third i added the schedulerid to pass the value to $SCHEDULER_ID variable.
Indeed the new insert for script scheduling is

`insert into scheduler(id,active,interval_ms,filename,arg1,arg2,arg3) values (11,1,'5000','/usr/bin/proxysql_node_monitor',10,'/var/lib/proxysql/proxysql_node_monitor_default.log',11);`